### PR TITLE
Clarinet-left-nav-changes

### PR DESCRIPTION
**Pull request by LakshmiLavanyaKasturi**
**DONOT merge this PR** This PR is created only for review.

Clarinet left nav changes. 

Files changed:
- sidebar.js
- docs/clarinet/

Related to https://github.com/hirosystems/docs/issues/138.

### CODE REVIEW

File: docs/clarinet/developer-resources.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+# Contributing to Clarinet
+
+We welcome contributions to Clarinet! The following sections provide information on how to contribute.
+
+## Prerequisites
+
+- rust (>=1.52.0)
+- cargo (>=1.52.0)
+- node (>=v14.16.0) - Used for git commit hook
```
Comment (obycode) [inline docs/clarinet/developer-resources.md:13]: Is this correct? Does clarinet have a git commit hook that requires node/npm?

File: docs/clarinet/faq.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+![FAQ -1](images/clarinet-faq-2.png)
+
+![FAQ -1](images/clarinet-faq-3.png)
+
+![FAQ -1](images/clarinet-faq-4.png)
+
+
+- `clarinet integrate` is not working
+
+check for rootless docker installation
```
Comment (obycode) [inline docs/clarinet/faq.md:29]: This content I added into the Google doc was just notes. It needs to be properly re-worded and formatted for a FAQ.

File: docs/clarinet/feature-guides.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
+---
+title: Feature Guides
```
Comment (obycode) [inline docs/clarinet/feature-guides.md:2]: I'm a bit unclear on this section, "Feature Guides". This does not seem like the correct content for this section.

File: docs/clarinet/getting-started.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
+---
+# The default id is the same as the one being defined below. so not needed
+title: Get started with Clarinet
+---
+
+# Getting started with Clarinet
+
+Before creating your project, make sure that you have installed and built clarinet by referring to [how-to install and build Clarinet](how-to-guides/how-to-install-clarinet.md).
+
+Now that you have installed and built Clarinet, you can create a new project and then populate the project with smart contracts.
```
Comment (obycode) [inline docs/clarinet/getting-started.md:10]: Should there be links to the how-tos for creating a new project, add contract, etc. here? This content is not that helpful on its own for a getting started section.

File: docs/clarinet/how-to-guides/how-to-add-contract.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+│   └── Testnet.toml
+└── tests
+    └── bbtc_test.ts
+```
+
+Clarinet will also add a configuration to the `Clarinet.toml` file for your contract.
+
+```toml
+[project.cache_location]
+path = ".requirements"
```
Comment (obycode) [inline docs/clarinet/how-to-guides/how-to-add-contract.md:34]: I don't think these two lines should be included here. I see that this was copied from our README and should be fixed there as well.

File: docs/clarinet/how-to-guides/how-to-analyse-with-check-checker.md [hunk 6]
```diff
@@ -0,0 +1,176 @@
+---
+title: Analyse with Check-Checker
```
Comment (obycode) [inline docs/clarinet/how-to-guides/how-to-analyse-with-check-checker.md:2]:
```suggestion
title: Analyze with Check-Checker
```
🇺🇸 😄 , but also, for consistency with the rest of the doc.
Perhaps change the filename to match as well.

File: docs/clarinet/how-to-guides/how-to-check-contract.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+This command uses the `Clarinet.toml` file to locate and analyze all of the contracts in the project.
+If the Clarity code is valid, then the command will indicate success with the response below.
+
+```
+✔ 2 contracts checked
+```
+
+The command may also report warnings indicating the code is valid, but there is something that you should pay attention to.
+For example, the check-checker analysis described below will generate warnings. If there are errors in the code,
```
Comment (obycode) [inline docs/clarinet/how-to-guides/how-to-check-contract.md:21]: > described below

Will this still be accurate in the new layout, or should it be reworded?

File: docs/clarinet/how-to-guides/how-to-create-new-project.md [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+│   └── Testnet.toml
+│   └── Mainnet.toml
+└── tests
+```
+
+The `Clarinet.toml` file contains configuration files for the smart contracts in your project. When you create contracts in
+your project, Clarinet will add them to this file.
+
+The `settings/Devnet.toml` file contains configuration for accounts in the Clarinet console, including the seed
+phrases and initial balances. Initial balances are in Micro Stacks (microSTX).
```
Comment (obycode) [inline docs/clarinet/how-to-guides/how-to-create-new-project.md:29]: > Micro Stacks (microSTX)

I believe the more common way of writing these would be "microstacks (uSTX)"

File: docs/clarinet/how-to-guides/how-to-monitor-test-coverage.md [hunk 9]
```diff
@@ -0,0 +1,7 @@
+---
+title: Monitor Test Coverage
+---
+
+You can measure and increase the code coverage by following the steps mentioned [here](https://github.com/hirosystems/clarinet/#measure-and-increase-code-coverage).
```
Comment (obycode) [inline docs/clarinet/how-to-guides/how-to-monitor-test-coverage.md:5]: Should these steps just be copied inline here instead of linking to another place?

File: docs/clarinet/how-to-guides/how-to-test-contract.md [hunk 10]
```diff
@@ -0,0 +1,255 @@
...
+            contract-name: bitcoin-whales
+            expected-sender: ST1PQHQKV0RJXZFY1DGX8MNSNYVE3VGZJSRTPGZGM
+            cost: 87210
+            path: "contracts\\bitcoin-whales.clar"
+            anchor-block-only: true
+```
+
+As you can see from the example above, Clarinet will remap the external contract to our Devnet address. In addition, Clarinet will also create a copy of it in the folder `requirements`
+
+## Deploy contracts to Devnet / Testnet / Mainnet
```
Comment (obycode) [inline docs/clarinet/how-to-guides/how-to-test-contract.md:211]: This section seems like it should be on its own page, since deploying is not related to testing.

File: docs/clarinet/how-to-guides/how-to-monitor-test-coverage.md [hunk 11]
```diff
@@ -0,0 +1,7 @@
+---
+title: Monitor Test Coverage
```
Comment (obycode) [inline docs/clarinet/how-to-guides/how-to-monitor-test-coverage.md:2]: Should this just be a section in the `how-to-test-contract.md` file?

File: docs/clarinet/introduction.md [hunk 12]
```diff
@@ -0,0 +1,17 @@
+---
+# The default id is the same as the one being defined below. so not needed
+title: What is Clarinet?
+---
+
+## What is Clarinet?
+
+Clarinet is a Clarity runtime packaged as a command line tool, designed to facilitate smart contract understanding, development, testing and deployment. Clarinet consists of a Clarity REPL and a testing harness, which, when used together allow you to rapidly develop and test a Clarity smart contract, with the need to deploy the contract to a local devnet or testnet.
```
Comment (obycode) [inline docs/clarinet/introduction.md:8]: Repeating my comment from the other place this paragraph shows up: I think it is a bit outdated and needs to be enhanced to include more recent features. I think this was written in early clarinet days.

**Review by obycode — CHANGES_REQUESTED** [review]
This is a good start. I added some comments.

File: docs/clarinet/introduction.md [hunk 13]
```diff
@@ -0,0 +1,17 @@
+---
+# The default id is the same as the one being defined below. so not needed
+title: What is Clarinet?
+---
+
+## What is Clarinet?
```
Comment (sabbyanandan) [inline docs/clarinet/introduction.md:6]: For consistency, can we rephrase this to "Clarinet overview"? All products can follow this as a pattern.

File: docs/clarinet/developer-resources.md [hunk 14]
```diff
@@ -0,0 +1,81 @@
+---
+title: Developer Resources
+---
+
+# Contributing to Clarinet
```
Comment (sabbyanandan) [inline docs/clarinet/developer-resources.md:5]: I understand this is copied over from the Clarinet README in Git. But the instructions here refer to "this" repo at a few places, which needs polishing.

File: docs/clarinet/faq.md [hunk 15]
```diff
@@ -0,0 +1,29 @@
+---
+title: FAQ
```
Comment (sabbyanandan) [inline docs/clarinet/faq.md:2]: It'd be better to see FAQs either numbered or indented, so it is clear for the readers. I cannot tell what this section specifically speaks about, also. Perhaps follow a Q+A like format, so we can ask a question and slide in the answer with screenshots?

File: docs/clarinet/feature-guides.md (hunk 3 above)
Comment (sabbyanandan) [reply docs/clarinet/feature-guides.md:2]: +1 to the observation. The helper could be considered a utility, but not a prominent ones like others.

Features could be all the nouns and verbs in Clarinet CLI, which we could describe or point to its CLI-level docs. Debugger is a feature. Devnet (aka: `clarinet integrate`) is another. Deployment plans, etc. 

These are likely captured in how-to's, so we can compose a top-level summary of all the features here or cross-link to relevant sections.

File: docs/clarinet/how-to-guides/how-to-install-clarinet.md [hunk 16]
```diff
@@ -0,0 +1,99 @@
...
+
+## Install on macOS (Homebrew)
+
+To install Clarinet using macOs, you must first have Homebrew installed on your system. If you do not already have Homebrew already installed, 
+please refer to the [Homebrew](https://brew.sh/)documentation for detailed information on how to install Homebrew.
+
+Once you have Homebrew installed, run the following command shown below in your terminal.
+
+```bash
+brew install clarinet
```
Comment (sabbyanandan) [inline docs/clarinet/how-to-guides/how-to-install-clarinet.md:25]: These appear in the s"setting up local environment"; perhaps remove one of these sections altogether. Whichever section comes first in the left-nav (ordering-wise), that wins and the other can be nuked.

**Review by sabbyanandan — CHANGES_REQUESTED** [review]

File: docs/clarinet/faq.md [hunk 17]
```diff
@@ -0,0 +1,45 @@
+---
+title: FAQ's
+---
+
+### Question - 1:
```
Comment (sabbyanandan) [inline docs/clarinet/faq.md:5]: Similar to the consolidation and conventions comments, we could follow this Q +A format for all products. Stacks.js structure in FAW looks different compared to this format.

File: docs/clarinet/getting-started.md [hunk 18]
```diff
@@ -0,0 +1,27 @@
+---
+# The default id is the same as the one being defined below. so not needed
+title: Get started with Clarinet
```
Comment (sabbyanandan) [inline docs/clarinet/getting-started.md:3]: "Get started with Clarinet" or simply "Get started" - let's stick to a convention, please.

File: docs/clarinet/developer-resources.md [hunk 19]
```diff
@@ -0,0 +1,66 @@
...
+      for testdir in $(ls examples); do
+          pushd examples/${testdir}
+              ../../target/debug/clarinet test .
+          popd
+      done
+      ```
+1. Submit a pull request against the `develop` branch for review.
+
+
+For more information, refer to the following links.
```
Comment (sabbyanandan) [inline docs/clarinet/developer-resources.md:61]: Should these links be grouped under a blog section or similar?

**Review by sabbyanandan — CHANGES_REQUESTED** [review]

**Comment by sabbyanandan** [comment]
Addressed in https://github.com/hirosystems/docs/commit/2ac6d6e8c01c18f6b11a197e70467b519bf9e0f5.
